VZ-8115: Pod Security and container security for Kiali

pshirodkar-orcl opened the pull request:
Pod Security and container security for Kiali.

pshirodkar-orcl commented on platform-operator/helm_config/overrides/kiali-server-values.yaml:
```diff
@@ -18,6 +18,22 @@ deployment:
     enabled: false
   instance_name: vmi-system-kiali
   view_only_mode: true
+  pod_security_context:
```
Before: Pod security warning entry for kiali can be seen.
![Screenshot 2023-01-12 at 3 49 04 PM](https://user-images.githubusercontent.com/93555077/212410726-dfe6c37c-a5bb-4722-a19a-4aafdf01168c.png)

After: Pod security warning entry for kiali is no longer seen.
<img width="1440" alt="Screenshot 2023-01-13 at 12 13 41 PM" src="https://user-images.githubusercontent.com/93555077/212411061-c18680fa-02b5-412d-ae17-737877fb1dd5.png">

mbenmclean reviewed (changes requested):
modify pod_security_test.go skipPods variable to ensure that kiali gets tested. Make the vmi-system prefix more specific so it doesn't catch Kiali and Kiali gets tested.

pshirodkar-orcl commented:
> modify pod_security_test.go skipPods variable to ensure that kiali gets tested. Make the vmi-system prefix more specific so it doesn't catch Kiali and Kiali gets tested.

I have addressed your comment now.
